target/linux/ramips/dts/mt7620a_asus_rt-ac54u.dts

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "asus,rt-ac54u", "ralink,mt7620a-soc";
+	model = "Asus RT-AC54U";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "rt-ac54u:blue:power";
+			gpios = <&gpio0 9 GPIO_ACTIVE_LOW>;
+		};
+
+		usb {
+			label = "rt-ac54u:blue:usb";
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&ohci_port1>, <&ehci_port1>;
+			linux,default-trigger = "usbport";
+		};
+
+		wifi {
+			label = "rt-ac54u:blue:wifi";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0{
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xfb0000>;
+			};
+		};
+	};
+};
+
+&ehci {
+	status = "okay";
+};
+
+&ohci {
+	status = "okay";
+};
+
+&gpio0 {
+	enable-leds {
+		gpio-hog;
+		line-name = "enable-leds";
+		output-low;
+		gpios = <10 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&ethernet {
+	status = "okay";
+	mtd-mac-address = <&factory 0x4>;
```

Comment:
Is there a proper address in 0x28, 0x2e, 0xe000 or 0xe006?

Reply:
0x4, 0x22, 0x28 have addresses.

Reply:
Then please give an overview over the addresses in flash and the addresses on vendor firmware, so we can assign them properly.
FYI: https://openwrt.org/docs/guide-developer/mac.address

Reply:
factory 0x4, 0x28: WAN, 2.4G (both addresses have the same value)
factory 0x22: LAN, 5G

Reply:
Ah, perfect. Then, please use the 0x22 for &ethernet and the 0x28 for wan_mac in the lower part of 02_network (instead of the increment). The 5g address should also be in 0x8004, as this is what's read automatically with mediatek,mtd-eeprom, like 0x4 is automatically read for 2g ...

Reply:
Please also an overview to the commit message, something like this:

```
MAC assignment based on vendor firmware:
2g    0x4         label + 1
5g    0x8004   label
lan    0x22      label
wan  0x28      label + 1
```

The third column should be adjusted so it's actually correct (can't deduce that from the information provided so far). If you device doesn't have a label with a particular address on it, just define one address as base address for the overview instead (and call it base or whatever)

Reply:
Oh yes you are right. 5G address is in 0x8004

Reply:
@adrianschmutzler label address is the same as LAN address

Reply:
I don't understand why the label +1?

Reply:
Typically, MAC addresses assigned to a device are subsequent. This is just another kind of information for somebody looking at this in the future:

label: aa:bb:cc:dd:ee:01
"label + 1": aa:bb:cc:dd:ee:02
"label - 1": aa:bb:cc:dd:ee:00
etc.

So the reader won't only know which address is where, but also their offset with respect to each other.

Reply:
However, I don't know which "offset" is correct with respect to the label in your case, so just see my table as a blueprint.

Reply:
If your label shows LAN address, then you should add `label-mac-device = &ethernet;` to the aliases section.

Reply:
LAN: aa:bb:cc:dd:ee:6c
WAN: aa:bb:cc:dd:ee:68
so the offset should be -4?

Reply:
yes, exactly.

Reply:
I changed as above now 5G doesn't get the right address

Reply:
Maybe just push your latest state so I can have a look.